Thomas peer

trobe137 opened the pull request:
I tentatively say that the peer code is finished. It is worth noting:
1.  Each peer only works with one file.
2. The Receiving peer will try to open the file.
3. Message size = number of bytes in file

trobe137 committed:
ClientThread Recognizes 404NOTFOUND

when the client thread received 404NOTFOUND, it will exit the system

trobe137 committed:
Closes Readers/Writers

Reads in a final statement from PeerThread and terminates connection

trobe137 committed:
Opens file, Closes sockets

The PeerThread will try to open the file. It will send a return statement if the file was successfully opened. Then it will close sockets, terminating connection

trobe137 committed:
added data variables

mkrupczak3 commented on Peer/ClientThread.java:
```diff
@@ -55,6 +55,17 @@ public void run()
+        if(targetIP.contains("NOTFOUND"))
+        {
+            System.err.println("The target IP was not found.\nExiting Program.");
+            System.exit(1);
```
@trobe137 System.exit() will kill all threads. Replace this with something more appropriate, so that the receiving thread isn't killed?

mkrupczak3 commented on Peer/PeerThread.java:
```diff
@@ -44,18 +45,31 @@ public void run()
-            }
-            if(in != null)
+            File tempFile = new File(fileName);
+            if(Desktop.isDesktopSupported() && tempFile.exists())
```
This will fail when inside a docker container. 

Remove anything which can't be run headless

mkrupczak3 commented on Peer/PeerThread.java:
```diff
@@ -44,18 +45,31 @@ public void run()
             {
-                in.close();
+                Desktop myDesk = Desktop.getDesktop();
+                myDesk.open(tempFile);
```
Remove this

mkrupczak3 reviewed (changes requested):
@trobe137 Looking really good, I have just a couple of nits that will save us some trouble down the line

trobe137 committed:
Removed Exit statements from threads

Removed System.exit from the threads. I also removed the code to try to open the file.

trobe137 commented:
Okay, I removed the desktop functions and most exit statements that occurred in threads.

trobe137 committed:
removed import desktop import statement
